modules/integration-core/src/test/scala/scala/cli/integration/RunDockerTests.scala

```diff
@@ -0,0 +1,45 @@
+package scala.cli.integration
+
+import com.eed3si9n.expecty.Expecty.expect
+
+import java.io.{ByteArrayOutputStream, File}
+import java.nio.charset.Charset
+
+import scala.util.Properties
+
+class RunDockerTests extends munit.FunSuite {
+
+    def simplRunAppDocker(): Unit = {
+        val fileName = "simple.sc"
+        val message = "Hello"
+        val inputs = TestInputs(
+            Seq(
+                os.rel / fileName ->
+                s"""val msg = "$message"
+                    |println(msg)
+                    |""".stripMargin
+            )
+        )
+        inputs.fromRoot { root =>
+            val termOpt = if (System.console() == null) Nil else Seq("-t")
+            val rawOutput = new ByteArrayOutputStream
+            os.proc("docker", "run", "--rm", termOpt, "-v", s"$root:/data", "-w", "/data", "scala-cli", fileName).call(
+                cwd = root,
+                stdout = os.ProcessOutput { (b, len) =>
+                rawOutput.write(b, 0, len)
+                System.err.write(b, 0, len)
+                },
+                mergeErrIntoOut = true
+            )
+            val output = new String(rawOutput.toByteArray, Charset.defaultCharset())
+            expect(output.linesIterator.toVector.last == message)
+        }
+    }
+
+    if (Properties.isLinux && TestUtil.cliKind == "docker")
```

Comment:
We shouldn't need this check… We'll only run these tests on systems that support it. (Incidentally, running them on macOS or Windows might work, as the mostly static binary is built from docker too.)